pipeline/trusted-resources/cmd/sign/sign_test.go

```diff
@@ -77,44 +86,81 @@ var (
 	}
 )
 
-func TestSignTask(t *testing.T) {
+func TestSign(t *testing.T) {
 	ctx := context.Background()
 
 	sv, err := trustedtask.GetSignerVerifier(password)
 	if err != nil {
 		t.Fatalf("error get signerverifier: %v", err)
 	}
 
-	var writer bytes.Buffer
-	ts := getTask()
-	if err := SignTask(ctx, ts, sv, &writer); err != nil {
-		t.Fatalf("Sign() get err %v", err)
+	tcs := []struct {
+		name     string
+		resource resourcesemantics.GenericCRD
+		kind     string
+	}{{
+		name:     "Task Sign and pass verification",
+		resource: getTask(),
+	}, {
+		name:     "Pipeline Sign and pass verification",
+		resource: getPipeline(),
+	},
 	}
 
-	signed := writer.Bytes()
-	ts, signature := unmarshalTask(t, signed)
-
-	if err := trustedtask.VerifyInterface(ctx, ts, sv, signature); err != nil {
-		t.Fatalf("VerifyTaskOCIBundle get error: %v", err)
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			var writer bytes.Buffer
+			if err := Sign(ctx, tc.resource, sv, &writer); err != nil {
+				t.Fatalf("Sign() get err %v", err)
+			}
+			signed := writer.Bytes()
+			target, signature := unmarshal(t, signed, tc.resource.GetObjectKind().GroupVersionKind().Kind)
+			if err := trustedtask.VerifyInterface(ctx, target, sv, signature); err != nil {
+				t.Fatalf("VerifyTaskOCIBundle get error: %v", err)
+			}
+		})
 	}
-
 }
 
 // unmarshalTask will get the task from buffer extract the signature.
-func unmarshalTask(t *testing.T, buf []byte) (*v1beta1.Task, []byte) {
-	task := &v1beta1.Task{}
-	if err := yaml.Unmarshal(buf, &task); err != nil {
-		t.Fatalf("error unmarshalling buffer: %v", err)
+func unmarshal(t *testing.T, buf []byte, kind string) (interface{}, []byte) {
```

Comment:
maybe this func can be cleaner

Reply:
unmarshalCRD?

Reply:
sure!

Reply:
previously I return interface{}, but it make more sense to return crd at this moment